tests/integration-tests/tests/common/assertions.py

```diff
@@ -249,6 +256,43 @@ def assert_aws_identity_access_is_correct(cluster, users_allow_list, remote_comm
         assert_that(result.failed).is_equal_to(not allowed)
 
 
+def assert_default_user_has_desired_sudo_access(
+    cluster, node_type, disable_sudo_access_default_user, region, scheduler_commands_factory, reboot_nodes=False
+):
+    remote_command_executors = []
+    if reboot_nodes:
+        logging.info(f"Rebooting nodes in the cluster {cluster.name}")
+        scheduler_commands = scheduler_commands_factory(RemoteCommandExecutor(cluster))
+        reboot_head_node(cluster)
+        reboot_login_node(cluster)
+        for partition in scheduler_commands.get_partitions():
+            compute_nodes = scheduler_commands.get_compute_nodes(filter_by_partition=partition)
+            logging.info(f"Rebooting compute nodes: {compute_nodes} in partition {partition}")
+            for compute_node in compute_nodes:
+                scheduler_commands.reboot_compute_node(compute_node, asap=False)
+            scheduler_commands.wait_nodes_status("idle", compute_nodes)
+            logging.info(f"Compute nodes in partition {partition} now IDLE: {compute_nodes}")
+
+    logging.info(
+        f"Asserting sudo access is {'disabled' if disable_sudo_access_default_user else 'enabled'} "
+        + f"for default user in node {node_type.value}"
+    )
+    if node_type == NodeType.HEAD_NODE:
+        remote_command_executors.append(RemoteCommandExecutor(cluster))
+    elif node_type == NodeType.COMPUTE_NODES:
+        for compute_node_ip in get_compute_nodes_instance_ips(cluster.name, region):
+            remote_command_executors.append(RemoteCommandExecutor(cluster, compute_node_ip=compute_node_ip))
+    elif node_type == NodeType.LOGIN_NODES:
+        remote_command_executors.append(RemoteCommandExecutor(cluster, use_login_node=True))
+
+    command = "sudo cat /etc/sudoers.d/90-cloud-init-users"
+    for node_index, remote_command_executor in enumerate(remote_command_executors):
+        result = remote_command_executor.run_remote_command(command, raise_on_error=False)
+        logging.info(f"Default user in {node_type} number {node_index} and result.failed={result.failed}")
```

Comment:
Checking that the command failed is not enough as it could fail for many reasons. We should specifically verify that the command fails with the expected error

Reply:
It asks for password for the user with a message which differs among the OSSes. Will check if it asks for password.